### PR TITLE
timer: drop HV interface, use it less often

### PR DESCRIPTION
**Pull request by roman-khimov**
Take a look, suggestions are welcome. But I think it's easier to use this way.

### CODE REVIEW

File: context.go [hunk 1]
```diff
@@ -63,9 +69,9 @@ type Context[H Hash] struct {
 	ChangeViewPayloads []ConsensusPayload[H]
 	// LastChangeViewPayloads stores consensus ChangeView payloads for the last epoch.
 	LastChangeViewPayloads []ConsensusPayload[H]
-	// LastSeenMessage array stores the height of the last seen message, for each validator.
-	// if this node never heard from validator i, LastSeenMessage[i] will be -1.
-	LastSeenMessage []*HV
+	// LastSeenMessage array stores the height and view of the last seen message, for each validator.
+	// if this node never heard from validator i, LastSeenMessage[i] will be nil.
```
Comment (AnnaShaleva) [inline context.go:73]: New sentence, capitalize the first word.

File: dbft.go [hunk 2]
```diff
@@ -168,22 +168,22 @@ func (d *DBFT[H]) OnTransaction(tx Transaction[H]) {
 }
 
 // OnTimeout advances state machine as if timeout was fired.
-func (d *DBFT[H]) OnTimeout(hv HV) {
+func (d *DBFT[H]) OnTimeout(height uint32, view byte) {
```
Comment (AnnaShaleva) [inline dbft.go:171]: I also had this thought but for some reason I thought that you'll be against such changes. This way is simpler and LGTM.

File: context.go [hunk 3]
```diff
@@ -6,6 +6,12 @@ import (
 	"time"
 )
 
+// HeightView is a block height/consensus view pair.
+type HeightView struct {
```
Comment (AnnaShaleva) [inline context.go:10]: We need to adjust README wrt these changes.

**Review by AnnaShaleva — COMMENTED** [review]
Need to add a note to CHANGELOG. The PR itself LGTM.

File: context.go (hunk 1 above)
Comment (roman-khimov) [reply context.go]: Fixed.

File: context.go (hunk 3 above)
Comment (roman-khimov) [reply context.go:10]: Fixed.

**Review by AnnaShaleva — APPROVED** [review]

**Pull request merged**